src/Command/Push/PushArtifactCommand.php

```diff
@@ -0,0 +1,242 @@
+    $this->logger->info('Finding Drupal core text files');
+    $sanitizeFinder = Finder::create()
+      ->files()
+      ->name('*.txt')
+      ->notName('LICENSE.txt')
+      ->in("$artifact_dir/docroot/core");
+
+    $this->logger->info('Finding VCS directories');
+    $vcsFinder = Finder::create()
+      ->ignoreDotFiles(FALSE)
+      ->ignoreVCS(FALSE)
+      ->directories()
+      ->in(["$artifact_dir/docroot",
+        "$artifact_dir/vendor",
+      ])
+      ->name('.git');
+    $drush_dir = "$artifact_dir/drush";
+    if (file_exists($drush_dir)) {
+      $vcsFinder->in($drush_dir);
+    }
+    if ($vcsFinder->hasResults()) {
+      $sanitizeFinder->append($vcsFinder);
+    }
+
+    $this->logger->info('Finding INSTALL database text files');
+    $dbInstallFinder = Finder::create()
+      ->files()
+      ->in([$artifact_dir])
+      ->name('/INSTALL\.[a-z]+\.(md|txt)$/');
+    if ($dbInstallFinder->hasResults()) {
+      $sanitizeFinder->append($dbInstallFinder);
+    }
+
+    $this->logger->info('Finding other common text files');
+    $filenames = [
+      'AUTHORS',
+      'CHANGELOG',
+      'CONDUCT',
+      'CONTRIBUTING',
+      'INSTALL',
+      'MAINTAINERS',
+      'PATCHES',
+      'TESTING',
+      'UPDATE',
+    ];
+    $textFileFinder = Finder::create()
+      ->files()
+      ->in(["$artifact_dir/docroot"])
+      ->name('/(' . implode('|', $filenames) . ')\.(md|txt)$/');
+    if ($textFileFinder->hasResults()) {
+      $sanitizeFinder->append($textFileFinder);
+    }
+
+    $this->logger->info("Removing sanitized files from build");
+    $this->localMachineHelper->getFilesystem()->remove($sanitizeFinder);
+  }
+
+  protected function commit(Closure $output_callback, string $artifact_dir, string $commit_hash):void {
+    $this->logger->info('Adding and committing changed files');
+    $this->localMachineHelper->executeFromCmd('git add -A', $output_callback, $artifact_dir, $this->output->isVerbose());
+    foreach (self::vendorFiles() as $vendor_directory) {
+      // This will fatally error if the directory doesn't exist. Suppress error output.
+      $this->logger->debug("Forcibly adding $vendor_directory");
+      $this->localMachineHelper->executeFromCmd('git add -f ' . $vendor_directory, NULL, $artifact_dir, FALSE);
+    }
+    $this->localMachineHelper->executeFromCmd('git commit -m "Automated commit by Acquia CLI (source commit: ' . $commit_hash . ')"', $output_callback, $artifact_dir, $this->output->isVerbose());
+  }
+
+  protected function push(Closure $output_callback, string $artifact_dir):void {
+    $this->logger->info('Pushing changes to Acquia Git');
+    $this->localMachineHelper->executeFromCmd('git push', $output_callback, $artifact_dir, $this->output->isVerbose());
+  }
+
+  private static function vendorFiles(): array {
+    return [
+      'vendor',
+      'docroot/core',
+      'docroot/modules/contrib',
+      'docroot/themes/contrib',
+      'docroot/profiles/contrib',
+      'docroot/libraries',
```

Comment:
Seems like we may want to read some of these values from composer.json.

Reply:
Great idea!